env0/resource_environment.go

```diff
@@ -0,0 +1,184 @@
+package env0
+
+import (
+	"context"
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceEnvironment() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceEnvironmentCreate,
+		ReadContext:   resourceEnvironmentRead,
+		UpdateContext: resourceEnvironmentUpdate,
+		DeleteContext: resourceEnvironmentDelete,
+
+		Importer: &schema.ResourceImporter{StateContext: resourceTeamImport},
+
+		Schema: map[string]*schema.Schema{
```

Comment:
only added basic fields, gonna add the rest in [part 2](https://trello.com/c/K68z3UXy/4784-terraform-provider-environment-resource-part-2)